### DISCUSSION TITLE
PyPDF2._page.PageObject._merge_page not working as expected

### DISCUSSION THREAD

**Discussion by rwu-iris** [py-pdf/pypdf#1593] [Q&A]
Dear PyPDF2 team, we have been using your library for our project, and it has been working well overall. Thank you team for the support all along. However, we have encountered this issue from time to time that when merging pages of some pdf file, one of the pages would not show. In all of these cases, it is the custom page that we create to overlay that is invisible.

I attach one example below. `test_page.pdf` is the target page and `test_canvas.pdf` is the file that we wish to overlay.
[test_page.pdf](https://github.com/py-pdf/pypdf/files/10556461/test_page.pdf)
[test_canvas.pdf](https://github.com/py-pdf/pypdf/files/10556462/test_canvas.pdf)

If you know anything regarding this issue, we would be very happy to learn :)

**Comment by pubpub-zz** [comment]
This could be due to  Number of digits.
PR #1563 is under final review that may solve your issue.
You may be interested also in #1567 which is also under finalization.
Can you try them to confirm that they will fix your issue?
